### PR TITLE
Pad the config.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/firmware.toit [hunk 1]
```diff
@@ -732,6 +732,10 @@ extract-host parsed/cli.Parsed envelope/Envelope --config-encoded/ByteArray:
   part-start = bits.size
   // Add the size to have a similar layout to the ESP32 binary.
   bits.little-endian.write-uint32 config-encoded.size
+  // Pad the config to 4 KB. This makes it less likely that the header (which includes
+  // the config-part size) changes for different configurations.
+  // We use 'N', since that's a NOP in UBJSON.
+  padded := pad config-encoded (4 * 1024) --pad-value='N'
   bits.write config-encoded
```
Comment (kasperl) [inline tools/firmware.toit:739]:
```suggestion
  bits.write padded
```

Comment (floitsch) [reply tools/firmware.toit]: oops :)

File: tools/firmware.toit [hunk 2]
```diff
@@ -732,7 +732,11 @@ extract-host parsed/cli.Parsed envelope/Envelope --config-encoded/ByteArray:
   part-start = bits.size
   // Add the size to have a similar layout to the ESP32 binary.
   bits.little-endian.write-uint32 config-encoded.size
-  bits.write config-encoded
+  // Pad the config to 4 KB. This makes it less likely that the header (which includes
+  // the config-part size) changes for different configurations.
+  // We use 'N', since that's a NOP in UBJSON.
+  padded := pad config-encoded (4 * 1024) --pad-value='N'
```
Comment (kasperl) [inline tools/firmware.toit:738]: I don't think this is good enough on second thought. We need to reserve space even if config-encoded is zero length.

**Review by kasperl — APPROVED** [review]
Looks good!

**Pull request merged**